### ISSUE TITLE
Child::kill will not terminate children on Linux

### ISSUE THREAD

**Issue by cottnn** [rust-lang/rust#115241]
### Location

https://doc.rust-lang.org/std/process/struct.Child.html#method.kill

### Summary

Documentation for `Child::kill` in standard library should include a note that the signal won't be sent to any children of the process, only to the process itself (at least on Linux).

This can be confusing because people might expect that `Child::kill` will terminate the process completely, including it's children.

I think the proper way to kill the process _and_ it's children is to send the signal to the process group

https://linux.die.net/man/2/killpg
https://linux.die.net/man/2/kill

**Comment by the8472** [comment]
> This can be confusing because people might expect that Child::kill will terminate the process completely, including it's children.

Well, that's not possible in general anyway since a descendant process may detach itself from the tree to get reparented to the nearest reaper.

**Comment by cottnn** [comment]
> Well, that's not possible in general anyway since a descendant process may detach itself from the tree to get reparented to the nearest reaper.

That's true, however I was only referring to children that weren't detached from their parent the time of calling `Child::kill`
